xformers/triton/activations.py

```diff
@@ -68,7 +68,12 @@ def relu(x):
 
 @triton.jit
 def relu_grad(x):
-    return tl.where(x >= 0, 1.0, 0.0)
+    # ReLU is different from other activations
+    # in that it does not require the input to retrospectively compute its gradient
+    # here the input is the downstream gradient, and we return the upstream gradient directly
+    zero = 0.0
+    zero = zero.to(x.dtype)
```

Comment:
Should this be done for zero in `squared_relu_grad` as well?

Reply:
it's a bit confusing, but it works without that because squared_relu ends up getting the saved activation input, which is fp32 by default (whereas with AMP the downstream gradient will be fp16). It makes me think that maybe there's some place to save here by casting it to fp16 when the inputs are fp16, will also improve the perf a little, that's a good point !